siliconcompiler/report/Home.py

```diff
@@ -0,0 +1,601 @@
+    if logs_and_reports == []:
+        return []
+    starting_path_name = logs_and_reports[0][0]
+    # reverse the list to start building the tree from the leaves up
+    for path_name, folders, files in reversed(logs_and_reports):
+        children = []
+        for folder in folders:
+            # assert (folder in subsect_logs_and_reports)
+            children.append(subsect_logs_and_reports[folder])
+        for file in files:
+            node = {}
+            node['label'] = file
+            # TODO: change to accomodate different operating systems
+            node['value'] = f'{path_name}/{file}'
+            children.append(node)
+        if starting_path_name == path_name:
+            return children
+        else:
+            node = {}
+            # TODO: change to accomodate different operating systems
+            folder = path_name[path_name.rfind('/') + 1:]
+            node['label'] = folder
+            node['value'] = path_name
+            node['children'] = children
+            subsect_logs_and_reports[folder] = node
+
+
+def get_nodes_and_edges(chip, node_dependencies, successful_path):
+    """
+    Returns the nodes and edges required to make a streamlit_agraph.
+
+    Args:
+        chip (Chip) : the chip object that contains the schema read from
+        node_dependencies (dict) : dictionary where the values of the keys are
+                                   the edges
+        successful_path (set) : contains all the nodes that are part of the
+                                'winning' path
+    """
+    nodes = []
+    edges = []
+    opacity = 0.2
+    width = 1
+    for step, index in node_dependencies:
+        if (step, index) in successful_path:
+            opacity = 1
+            if (step, index) in chip._get_flowgraph_exit_nodes() or \
+               (step, index) in chip._get_flowgraph_entry_nodes():
+                width = 3
+
+        flow = chip.get("option", "flow")
+        task_status = chip.get('flowgraph', flow, step, index, 'status')
+        if task_status == TaskStatus.SUCCESS:
+            node_color = success_color
+        elif task_status == TaskStatus.ERROR:
+            node_color = failure_color
+        else:
+            node_color = pending_color
+
+        tool, task = chip._get_tool_task(step, index)
+        label = step + index + "\n" + tool + "/" + task
+        if chip._is_builtin(tool, task):
+            label = step + index + "\n" + tool
+
+        nodes.append(Node(id=step + index,
+                          label=label,
+                          color=node_color,
+                          opacity=opacity,
+                          borderWidth=width,
+                          shape='oval'))
+        width = 3
+        for source_step, source_index in node_dependencies[step, index]:
+            if (source_step, source_index) in successful_path and \
+               (source_step, source_index) in successful_path:
+                width = 5
```

Comment:
magic number

Reply:
I've oscillated between making these inputs with defaults versus constants. I'm leaning towards making them inputs with defaults since these constants are only used in this function.